### PR TITLE
improve presence UX

### PR DESCRIPTION
**Pull request by caneleex**
this PR changes presence data to be optional, resulting in not having to pass unwanted data.

before:
```go
bot.WithGatewayConfigOpts(
			gateway.WithIntents(gateway.IntentsNone),
			gateway.WithPresence(gateway.NewListeningPresence("music", discord.OnlineStatusOnline, false)),
)
```
after:
```go
// this yields the same result as the example above
bot.WithGatewayConfigOpts(
			gateway.WithIntents(gateway.IntentsNone),
			gateway.WithPresenceOpts(gateway.WithListeningActivity("music")),
)
```

### CODE REVIEW

**Review by Skye-31 — APPROVED** [review]

File: gateway/gateway_config.go [hunk 1]
```diff
@@ -60,6 +60,8 @@ func (c *Config) Apply(opts []ConfigOpt) {
 	}
 }
 
+type PresenceOpt func(presenceUpdate *MessageDataPresenceUpdate)
```
Comment (TopiSenpai) [inline gateway/gateway_config.go:63]: one minor thing
Can you move this into `gateway/gateway_messages.go`?

File: gateway/gateway_messages.go [hunk 2]
```diff
@@ -437,57 +437,67 @@ type IdentifyCommandDataProperties struct {
...
-func NewCompetingPresence(name string, status discord.OnlineStatus, afk bool) MessageDataPresenceUpdate {
-	return NewPresence(discord.ActivityTypeCompeting, name, "", status, afk)
+func WithOnlineStatus(status discord.OnlineStatus) PresenceOpt {
+	return func(presence *MessageDataPresenceUpdate) {
+		presence.Status = status
+		if status == discord.OnlineStatusIdle {
+			var since *int64
+			unix := time.Now().Unix()
+			since = &unix
+			presence.Since = since
```
Comment (TopiSenpai) [inline gateway/gateway_messages.go:492]:
```suggestion
			since := time.Now().Unix()
			presence.Since = &since
```

**Review by TopiSenpai — CHANGES_REQUESTED** [review]
atm all fields in `MessageDataPresenceUpdate` are not optional. like this all fields get overwritten regardless. 
are they optional(discord docs don't say so)? we need to check else this pr is kinda pointless

File: gateway/gateway_messages.go [hunk 3]
```diff
@@ -437,57 +436,77 @@ type IdentifyCommandDataProperties struct {
...
-			Name: name,
-			Type: activityType,
-		}
-		if activityType == discord.ActivityTypeStreaming && url != "" {
-			activity.URL = &url
-		}
-		activities = append(activities, activity)
-	}
+// WithPlayingActivity creates a new "Playing ..." activity of type discord.ActivityTypeGame
+func WithPlayingActivity(name string) PresenceOpt {
```
Comment (TopiSenpai) [inline gateway/gateway_messages.go:442]: should we rename those functions to `WithActivityX` for better discoverability?

Comment (caneleex) [reply gateway/gateway_messages.go:442]: I think it's safe to assume people these days use intelligent IDEs which will already suggest the methods when using proper keywords such as "activity" or "listening".

**Review by TopiSenpai — APPROVED** [review]

**Pull request merged**